sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs

```diff
@@ -205,6 +207,39 @@ public async Task<HttpResponseMessage> ConvertAsync(HttpRequestMessage input, Ca
             return (T)blob.BlobClient;
         }
 
+        private ParameterBindingData ConvertToParameterBindingData(BlobAttribute blobAttribute)
+        {
+            var blobPath = BlobPath.ParseAndValidate(blobAttribute.BlobPath);
+
+            var blobDetails = new BlobParameterBindingDataContent()
+            {
+                Connection = _nameResolver.Resolve(blobAttribute.Connection) ?? Constants.DefaultAzureStorageConnectionName,
```

Comment:
The main point I was making previously here is that there's existing connection name prefixing logic that is used when resolving that customers will expect to work. E.g. follow the code from [here](https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/extensions/Microsoft.Azure.WebJobs.Extensions.Clients/src/Shared/StorageClientProvider.cs#L59) down, and you'll see the logic that will apply AzureWebJobs prefix to provided names if needed to resolve them. That means customers can use names like "Storage", or "Data" in their bindings and have corresponding env settings like "AzureWebJobsStorage", "AzureWebJobsData" and they'll resolve.

Because it'll now be the worker code that resolves these settings by name, we'll want to attempt to discover the correct value to use here I think, rather than duplicate that logic in the worker extension.

E.g. code like this, extracted into a helper somewhere:

```csharp
string connectionName = !string.IsNullOrEmpty(blobAttribute.Connection) ? resolver.ResolveWholeString(blobAttribute.Connection) : ConnectionStringNames.Storage;
IConfigurationSection connectionSection = _configuration.GetWebJobsConnectionStringSection(connectionName);
```

That would resolve to an actual configuration section which we can then use the name of. If it doesn't resolve, we know the connection is invalid, and we could throw like we do when resolving in proc bindings.

Reply:
Done/